racket/collects/syntax/to-string.rkt

```diff
@@ -58,7 +58,19 @@
                (define c-paren-shape (syntax-property c 'paren-shape))
                (printf "~a" (or c-paren-shape #\())
                (set! col (+ col 1))
-               (map (loop init-line!) (syntax->list c))
+               (letrec ([se (syntax-e c)]
+                        [build-string-from-pair (lambda (sp)
```

Comment:
I personally find that the rightward drift here is really too much. Can't we use internal definitions instead of `letrec`? Something like:

```
(define se (syntax-e c))
(define (build-string-from-pair sp)
  (cond
    [...]
    ...))
(if (list? se)
    ...
    ...)
```

Reply:
Thanks for the review. You're right, that was a case of fullscreen tunnel vision.